### PR TITLE
Add MPRESS command line options

### PR DESCRIPTION
**Pull request by trlkly**
This adds the ability to specify command line options to MPRESS, both in the GUI and on the command line. I also modified the help text, and save the last used GUI option to the registry.

### CODE REVIEW

**Comment by trlkly** [comment]
I assume that -x is necessary and should not be disabled. I don't 100% understand what it means by "don't manage exceptions," but I assume you added that option for a reason.

**Comment by trlkly** [comment]
I have a version that incorporates the other pull requests, too (including making the new options save in an INI file when appropriate). Not sure what to do with that. 

I could commit it, and then you could pull all of it into the master branch.

**Comment by fincs** [comment]
Please do not integrate any other people's PRs into yours, as that would defeat the entire purpose of sending a PR and would make your PR difficult to review.

Also, MPRESS was completely abandoned by the original developer and is still available for download only because the AutoHotkey Foundation [hosts a mirror](https://autohotkey.com/mpress/mpress_web.htm) of the original website (which has been gone for some time now). For this reason, as well as the fact that MPRESS compressed executables are highly likely to be picked up as false positives by AV software, I'm thinking about dropping support for automatic executable compression as a whole. I'd like to hear people argue whether this should be done or not.

**Comment by trlkly** [comment]
The problem is, my pull request will not work once you apply the other ones, since I touch on some of the same parts. In particular, I add to the registry entries. If you pull the INI file one, mine will fail to work properly. 

There needs to be some place that has all the latest features, and not just three completely separate versions. 

Speaking of not being maintained, why are there year-old open pull requests that have yet to be accepted or denied? The INI one sounds like it's already in use in the portable version, and the other one is a trivial feature that makes things easier with no downsides. Why have they not been checked in? 

As for removing MPRESS, I can see no reason to do so. Yes, the software isn't being developed or maintained, but it works. It doesn't need to be developed anymore. And it's not as if the script is doing any heavy lifting. It costs basically nothing to maintain the UI option(s). 

Yes, the AV thing can be a problem, but the problem is bad AVs that assume all compression is bad, rather than decompressing and checking. It's LZMA compression, for goodness sake: the same compression used in 7zip files. And it's intentionally fast to decrypt, or else it would suck for its purpose.

A good AV will not fire on the mere use of compressed EXEs. And the last thing to do is to let them be lazy and assume something useful as making EXE files smaller is inherently dangerous. Let them get all the false positive reports, and fix their software to work properly. 

Now, if you wanted to switch to a different compression method, that I might approve of. If there is one that is superior in compression and/or performance, that could be good. 

But having raw script in an EXE is silly.

File: Ahk2Exe.ahk [hunk 1]
```diff
@@ -280,6 +282,8 @@ return
 
 _ProcessMPRESS:
 UseMPRESS := p2
+if p2 && p2 != 1
+	MPRESSparams := p2
```
Comment (fincs) [inline Ahk2Exe.ahk]: If the /mpress value is different than 1, the value of UseMPRESS will be equal to the command line switches passed by the user. Likewise, if the value is 0, MPRESSparams will hold 0. UseMPRESS should always be boolean (0/1) and MPRESSparams should either be empty-string or ignored/unmodified if UseMPRESS is 0.

**Comment by joedf** [comment]
I was actually going to suggest that :)
________________________________

**Comment by joedf** [comment]
Interestingly enough, matcode.com is back online again :b
http://www.matcode.com/mpress.htm
________________________________

**Comment by joedf** [comment]
Superseded by ahk directives.
Mainly:
```ahk
;@Ahk2Exe-PostExec Command
```

**Pull request closed without merging**